bootstrap/kubeadm/internal/cloudinit/kubeadm-bootstrap-script.sh

```diff
@@ -129,8 +129,16 @@ retry-command kubeadm join phase control-plane-prepare control-plane
 # {{ end }}
 retry-command kubeadm join phase kubelet-start
 # {{ if .ControlPlane }}
+# {{ if .NewPhases  }}
+# Note: below phase is a no-op if ControlPlaneKubeletLocalMode feature gate is false.
+try-or-die-command kubeadm join phase etcd-join
+# {{ end }}
+# Note: below phase is a no-op if ControlPlaneKubeletLocalMode feature gate is false.
+retry-command kubelet-wait-boostrap
```

Comment:
IIRC, phase order was relevant to the FG enablement, do we need to consider it here?
(non trivial i would assume)

Reply:
Phase order is correct here, this is the file for >= v1.31

Reply:
there is a typo `boostrap`.

this is the 1.31 CLI output
```
preflight              Run join pre-flight checks
control-plane-prepare  Prepare the machine for serving a control plane
  /download-certs        Download certificates shared among control-plane nodes from the kubeadm-certs Secret
  /certs                 Generate the certificates for the new control plane components
  /kubeconfig            Generate the kubeconfig for the new control plane components
  /control-plane         Generate the manifests for the new control plane components
kubelet-start          Write kubelet settings, certificates and (re)start the kubelet
control-plane-join     Join a machine as a control plane instance
  /etcd                  Add a new local etcd member
  /mark-control-plane    Mark a node as a control-plane
wait-control-plane     EXPERIMENTAL: Wait for the control plane to start
```

but these etcd-join and kubelet-wait-bootstrap phases are hidden for now:
https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/cmd/kubeadm/app/cmd/phases/join/controlplanejoin.go#L93

https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/cmd/kubeadm/app/cmd/phases/join/kubelet.go#L103

so there has to be another iteration here. or you can use the fact that calling an unknown `kubeadm join phase non-existent` will return exit code 0, which is a bit of a mistake on the kubeadm side. just found it out by doing `kubeadm join phase foo > /dev/null && echo ok` and prepopulate these hidden phases.

Reply:
hmm, looks like I even cannot run these hidden phases manually :-(